src/stream/src/executor/aggregation/agg_impl/foldable.rs

```diff
@@ -194,6 +195,46 @@ where
     }
 }
 
+/// `BitXorable` returns the result of bit_xor all the values.
+/// It produces the same type of output as input `S`.
+#[derive(Debug)]
+pub struct BitXorable<S>
+where
+    S: Scalar + BitXor<Output = S>,
+{
+    _phantom: PhantomData<S>,
+}
+
+impl<S> StreamingFoldable<S, S> for BitXorable<S>
+where
+    S: Scalar + BitXor<Output = S>,
+{
+    fn accumulate(
+        result: Option<&S>,
+        input: Option<S::ScalarRefType<'_>>,
+    ) -> StreamExecutorResult<Option<S>> {
+        Ok(match (result, input) {
+            (Some(x), Some(y)) => Some(x.clone().bitxor(y.to_owned_scalar())),
```

Comment:
I suppose `clone` and `to_owned_scalar` is unavoidable under current abstraction?